src/gc/impl/conservative/gc.d

```diff
@@ -2581,9 +2604,54 @@ struct Gcx
         begin = start = currTime;
 
         debug(COLLECT_PRINTF) printf("Gcx.fullcollect()\n");
+        bool doFork = config.fork;
+        version (COLLECT_PARALLEL)
+            bool doParallel = config.parallel > 0;
+        else
+            enum doParallel = false;
+
+        assert(!(doFork && doParallel), "No reason in mixing threads and fork");
```

Comment:
this assert is dangerous because it will deadlock if the user runs with DRT-gcopt=fork:1 and doesn't disable threaded collection.

Every assert related to GC code deadlocks given that assert allocates. This is a problem related to code outside of this PR as well

Reply:
The assert will go away in a release build, emit an error instead.